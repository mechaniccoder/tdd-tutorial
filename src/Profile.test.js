import React from "react";
import {mount} from "enzyme";
import Profile from "./Profile";

describe("<Profile />", () => {
  it("matches snapshot", () => {
    const wrapper = mount(<Profile username="mechaniccoder" name="유승환" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders username and name", () => {
    const wrapper = mount(<Profile username="mechaniccoder" name="유승환" />);
    expect(wrapper.props().username).toBe("mechaniccoder");
    expect(wrapper.props().name).toBe("유승환");

    const boldElement = wrapper.find("b");
    expect(boldElement.text()).toBe("mechaniccoder");

    const spanElement = wrapper.find("span");
    expect(spanElement.text()).toBe("유승환");
  });
});
