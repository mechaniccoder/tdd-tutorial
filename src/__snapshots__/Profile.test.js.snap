// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> matches snapshot 1`] = `
<Profile
  name="유승환"
  username="mechaniccoder"
>
  <div>
    <b>
      mechaniccoder
    </b>
    <span>
      유승환
    </span>
  </div>
</Profile>
`;
